### PR TITLE
DMU 7 cards

### PR DESCRIPTION
**Pull request by Simisays**
@Hanmac Scripting my first Read Ahead card. Thought you might want to check the script

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/shadow_rite_priest.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Name:Shadow-Rite Priest
+ManaCost:1 B
+Types:Creature Human Cleric
+PT:2/2
+S:Mode$ Continuous | Affected$ Card.Cleric+Other+YouCtrl | AddPower$ 1 | AddToughness$ 1 | Description$ Other Clerics you control get +1/+1.
+A:AB$ ChangeZone | Cost$ T 3 B B Sac<1/Cleric.Other> | Origin$ Library | Destination$ Hand | ChangeType$ Creature.Black | SpellDescription$ Search your library for a black creature card, put it onto the battlefield, then shuffle.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/shadow_rite_priest.txt:6]: Wrong Destination

File: forge-gui/res/cardsfolder/upcoming/the_world_spell.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Name:The World Spell
+ManaCost:5 G G
+Types:Enchantment Saga
+K:Read ahead:3:DBDig,DBDig,DBChangeZone
+SVar:DBDig:DB$ Dig | DigNum$ 7 | ChangeNum$ 1 | Optional$ True | ForceRevealToController$ True | ChangeValid$ Card.nonSaga | RestRandomOrder$ True | SpellDescription$ Look at the top seven cards of your library. You may reveal a non-Saga permanent card from among them and put it into your hand. Put the rest on the bottom of your library in a random order.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_world_spell.txt:5]: `ChangeValid$ Permanent.nonSaga`

File: forge-gui/res/cardsfolder/upcoming/thran_portal.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Name:Thran Portal
+ManaCost:no cost
+Types:Land Gate
+K:ETBReplacement:Other:LandTapped
+SVar:LandTapped:DB$ Tap | Defined$ Self | ETB$ True | ConditionCheckSVar$ ETBCheckSVar2 | ConditionSVarCompare$ GT2 | SpellDescription$ CARDNAME enters the battlefield tapped unless you control two or fewer other lands.
+K:ETBReplacement:Other:DBChooseBasic
+SVar:DBChooseBasic:DB$ ChooseType | Type$ Basic Land | SpellDescription$ As CARDNAME enters the battlefield, choose a basic land type.
+S:Mode$ Continuous | Affected$ Card.Self | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+S:Mode$ RaiseCost | Type$ Ability | ValidCard$ Card.namedThran Portal | Cost$ PayLife<1> | EffectZone$ All | Description$ Activated abilities of CARDNAME cost an additional "1 life" to activate.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/thran_portal.txt:9]: `ValidCard$ Card.Self`

File: forge-gui/res/cardsfolder/upcoming/thran_portal.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
+Name:Thran Portal
+ManaCost:no cost
+Types:Land Gate
+K:ETBReplacement:Other:LandTapped
+SVar:LandTapped:DB$ Tap | Defined$ Self | ETB$ True | ConditionCheckSVar$ ETBCheckSVar2 | ConditionSVarCompare$ GT2 | SpellDescription$ CARDNAME enters the battlefield tapped unless you control two or fewer other lands.
+K:ETBReplacement:Other:DBChooseBasic
+SVar:DBChooseBasic:DB$ ChooseType | Type$ Basic Land | SpellDescription$ As CARDNAME enters the battlefield, choose a basic land type.
+S:Mode$ Continuous | Affected$ Card.Self | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+S:Mode$ RaiseCost | Type$ Ability | ValidCard$ Card.Self | Cost$ PayLife<1> | EffectZone$ All | Description$ Activated abilities of CARDNAME cost an additional "1 life" to activate.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/thran_portal.txt:9]: It should be mana abilities only

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/thran_portal.txt]: I think this required some engine changes, interested what you think about this

File: forge-game/src/main/java/forge/game/cost/CostAdjustment.java [hunk 5]
```diff
@@ -485,6 +485,10 @@ private static boolean checkRequirement(final SpellAbility sa, final StaticAbili
                 if (!sa.isActivatedAbility() || sa.isManaAbility() || sa.isReplacementAbility()) {
                     return false;
                 }
+            } else if (type.equals("ManaAbility")) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/cost/CostAdjustment.java:488]: This extension shouldn't be needed
Just try `ValidSpell$ Activated.ManaAbility` on the card instead

Comment (tool4ever) [reply forge-game/src/main/java/forge/game/cost/CostAdjustment.java]: You should not have dropped the ValidCard too

**Review by tool4ever — APPROVED** [review]

File: forge-gui/res/cardsfolder/upcoming/crystal_grotto.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+Name:Crystal Grotto
+ManaCost:no cost
+Types:Land 
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigScry | TriggerDescription$ When CARDNAME enters the battlefield, scry 1.
+SVar:TrigScry:DB$ Scry | ScryNum$ 1
+A:AB$ Mana | Cost$ T | Produced$ C | SpellDescription$ Add {C}.
+A:AB$ Mana | Cost$ 1 T | Produced$ Any | SpellDescription$ Add one mana of any color.
+AI:RemoveDeck:All
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/crystal_grotto.txt:8]: AI should be able to handle this?

File: forge-gui/res/cardsfolder/upcoming/drag_to_the_bottom.txt [hunk 7]
```diff
@@ -0,0 +1,6 @@
+Name:Drag to the Bottom
+ManaCost:2 B B 
+Types:Sorcery
+A:SP$ PumpAll | ValidCards$ Creature | NumAtt$ -X | NumDef$ -X | SpellDescription$ Domain — Each creature gets -X/-X until end of turn, where X is 1 plus the number of basic land types among lands you control.
+SVar:X:Count$Domain/Plus.1
+Oracle:Domain — Each creature gets -X/-X until end of turn, where X is 1 plus the number of basic land types among lands you control.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/drag_to_the_bottom.txt:6]: Add boilerplate Domain AI lines

File: forge-gui/res/cardsfolder/upcoming/shadow_rite_priest.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Name:Shadow-Rite Priest
+ManaCost:1 B
+Types:Creature Human Cleric
+PT:2/2
+S:Mode$ Continuous | Affected$ Card.Cleric+Other+YouCtrl | AddPower$ 1 | AddToughness$ 1 | Description$ Other Clerics you control get +1/+1.
+A:AB$ ChangeZone | Cost$ T 3 B B Sac<1/Cleric.Other> | Origin$ Library | Destination$ Battlefield | ChangeType$ Creature.Black | SpellDescription$ Search your library for a black creature card, put it onto the battlefield, then shuffle.
+DeckHas:Ability$Sacrifice 
+Oracle:Other Clerics you control get +1/+1.\n{3}{B}{B}, {T}, Sacrifice another Cleric: Search your library for a black creature card, put it onto the battlefield, then shuffle.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/shadow_rite_priest.txt:8]: DeckNeeds:Type$Cleric

File: forge-gui/res/cardsfolder/upcoming/shadow_rite_priest.txt [hunk 9]
```diff
@@ -0,0 +1,8 @@
+Name:Shadow-Rite Priest
+ManaCost:1 B
+Types:Creature Human Cleric
+PT:2/2
+S:Mode$ Continuous | Affected$ Card.Cleric+Other+YouCtrl | AddPower$ 1 | AddToughness$ 1 | Description$ Other Clerics you control get +1/+1.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/shadow_rite_priest.txt:5]: Cleric.Other+YouCtrl probably fine

**Review by tool4ever — APPROVED** [review]

**Review by tool4ever — DISMISSED** [review]

File: forge-gui/res/cardsfolder/upcoming/the_world_spell.txt [hunk 10]
```diff
@@ -0,0 +1,7 @@
+Name:The World Spell
+ManaCost:5 G G
+Types:Enchantment Saga
+K:Read ahead:3:DBDig,DBDig,DBChangeZone
+SVar:DBDig:DB$ Dig | DigNum$ 7 | ChangeNum$ 1 | Optional$ True | ForceRevealToController$ True | ChangeValid$ Permanent.nonSaga | RestRandomOrder$ True | SpellDescription$ Look at the top seven cards of your library. You may reveal a non-Saga permanent card from among them and put it into your hand. Put the rest on the bottom of your library in a random order.
+SVar:DBChangeZone:DB$ ChangeZone | Origin$ Hand | Destination$ Battlefield | ChangeType$ Permanent.nonSaga | ChangeNum$ 2 | SpellDescription$ Put up to two non-Saga permanent cards from your hand onto the battlefield.
+Oracle:Read ahead (Choose a chapter and start with that many lore counters. Add one after your draw step. Skipped chapters don’t trigger. Sacrifice after III.)\nI, II — Look at the top seven cards of your library. You may reveal a non-Saga permanent card from among them and put it into your hand. Put the rest on the bottom of your library in a random order.\nIII — Put up to two non-Saga permanent cards from your hand onto the battlefield.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/the_world_spell.txt:7]: don’t = don't
(curly)

File: forge-gui/res/cardsfolder/upcoming/thran_portal.txt [hunk 11]
```diff
@@ -0,0 +1,12 @@
+Name:Thran Portal
+ManaCost:no cost
+Types:Land Gate
+K:ETBReplacement:Other:LandTapped
+SVar:LandTapped:DB$ Tap | Defined$ Self | ETB$ True | ConditionCheckSVar$ ETBCheckSVar2 | ConditionSVarCompare$ GT2 | SpellDescription$ CARDNAME enters the battlefield tapped unless you control two or fewer other lands.
+K:ETBReplacement:Other:DBChooseBasic
+SVar:DBChooseBasic:DB$ ChooseType | Type$ Basic Land | SpellDescription$ As CARDNAME enters the battlefield, choose a basic land type.
+S:Mode$ Continuous | Affected$ Card.Self | AddType$ ChosenType | Description$ CARDNAME is the chosen type in addition to its other types.
+S:Mode$ RaiseCost | Type$ Ability | ValidSpell$ Activated.ManaAbility | ValidCard$ Card.Self | Cost$ PayLife<1> | EffectZone$ All | Description$ Mana abilities of CARDNAME cost an additional "1 life" to activate.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/thran_portal.txt:9]: "1 life"

the " shouldn't be here

**Review by paulsnoops — APPROVED** [review]

**Pull request merged**